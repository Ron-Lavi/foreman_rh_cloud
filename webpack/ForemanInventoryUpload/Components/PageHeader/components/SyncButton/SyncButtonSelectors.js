import { selectAPIResponse } from 'foremanReact/redux/API/APISelectors';
import { INVENTORY_SYNC_TASK_UPDATE } from './SyncButtonConstants';

export const selectTaskStatus = state => {
  const { result } = selectAPIResponse(state, INVENTORY_SYNC_TASK_UPDATE);
  return typeof result === 'string' ? result.toUpperCase() : null;
};
