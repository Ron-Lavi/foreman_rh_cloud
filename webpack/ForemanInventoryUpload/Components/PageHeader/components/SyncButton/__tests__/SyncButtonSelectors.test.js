import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import { selectTaskStatus } from '../SyncButtonSelectors';

const state = {
  API: {
    INVENTORY_SYNC_TASK_UPDATE: {
      response: {
        endedAt: '2021-03-08T14:27:30.718+02:00',
        output: {
          host_statuses: {
            sync: 0,
            disconnect: 2,
          },
        },
        result: 'pending',
      },
      status: 'RESOLVED',
    },
  },
};

const fixtures = {
  'should return InventorySync status': () => selectTaskStatus(state),
};

describe('SyncButton selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
