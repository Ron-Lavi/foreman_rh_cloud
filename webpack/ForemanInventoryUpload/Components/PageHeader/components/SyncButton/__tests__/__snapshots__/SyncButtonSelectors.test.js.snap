// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncButton selectors should return InventorySync status 1`] = `"PENDING"`;
